### ISSUE TITLE
BUG: rerender adds lots of empty changes to the git index on windows

### ISSUE THREAD

**Issue by h-vetinari** [conda-forge/conda-smithy#2191]
One less impactful but still annoying consequence of the pygit2 switch recently is that by default, when `conda smithy rerender` finishes, a bunch of stuff has been added to the index that's no change at all; e.g.
```
 .azure-pipelines/build_steps.sh                                         | 0
 .azure-pipelines/create_conda_build_artifacts.bat                       | 0
 .azure-pipelines/create_conda_build_artifacts.sh                        | 0
 .azure-pipelines/run_docker_build.sh                                    | 0
 .azure-pipelines/run_osx_build.sh                                       | 0
 .azure-pipelines/run_win_build.bat                                      | 0
 .../linux_64_is_freethreadingfalsenumpy1.22python3.10.____cpython.yaml  | 2 +-
[.... actual changes ...]
 .ci_support/win_64_is_freethreadingfalsenumpy2python3.13.____cp313.yaml | 2 +-
 .circleci/build_steps.sh                                                | 0
 .circleci/checkout_merge_commit.sh                                      | 0
 .circleci/create_conda_build_artifacts.bat                              | 0
 .circleci/create_conda_build_artifacts.sh                               | 0
 .circleci/fast_finish_ci_pr_build.sh                                    | 0
 .circleci/run_docker_build.sh                                           | 0
 .circleci/run_osx_build.sh                                              | 0
 .circleci/run_win_build.bat                                             | 0
 .drone/build_steps.sh                                                   | 0
 .drone/create_conda_build_artifacts.bat                                 | 0
 .drone/create_conda_build_artifacts.sh                                  | 0
 .drone/run_docker_build.sh                                              | 0
 .drone/run_osx_build.sh                                                 | 0
 .drone/run_win_build.bat                                                | 0
 .github/CONTRIBUTING.md                                                 | 0
 .github/ISSUE_TEMPLATE.md                                               | 0
 .github/PULL_REQUEST_TEMPLATE.md                                        | 0
 .github/workflows/automerge.yml                                         | 0
 .github/workflows/conda-build.yml                                       | 0
 .github/workflows/main.yml                                              | 0
 .github/workflows/webservices.yml                                       | 0
 .scripts/create_conda_build_artifacts.bat                               | 0
 .scripts/create_conda_build_artifacts.sh                                | 0
 .travis/build_steps.sh                                                  | 0
 .travis/create_conda_build_artifacts.bat                                | 0
 .travis/create_conda_build_artifacts.sh                                 | 0
 .travis/run_docker_build.sh                                             | 0
 .travis/run_osx_build.sh                                                | 0
 .travis/run_win_build.bat                                               | 0
 ci_support/checkout_merge_commit.sh                                     | 0
 ci_support/fast_finish_ci_pr_build.sh                                   | 0
 ci_support/run_docker_build.sh                                          | 0
 ci_support/upload_or_check_non_existence.py                             | 0
```

It's possible to reset the index and then commit, which then does not create all these spurious changes. It might be related to CRLF?

CC @mgorny

**Comment by mgorny** [comment]
Good news is that I was able to reproduce inside wine.

Bad news is that when I made a standalone test case for libgit2, I couldn't reproduce:

```c
#include <git2.h>

#include <assert.h>
#include <stdio.h>

int main() {
	git_repository *repo = NULL;
	git_index *idx = NULL;
	FILE *f;

	git_libgit2_init();
	assert(git_repository_open(&repo, ".") == 0);
	assert(git_repository_index(&idx, repo) == 0);

	f = fopen("foo.txt", "w");
	fputs("test\nothertest\nmoretest\n", f);
	fclose(f);

	assert(git_index_add_bypath(idx, "foo.txt") == 0);
	assert(git_index_write(idx) == 0);

	git_index_free(idx);
	git_repository_free(repo);
	git_libgit2_shutdown();

	return 0;
}
```

Also couldn't reproduce when mixing calls between Linux and Wine applications in the same directory (i.e. effectively shuffling between CRLF and LF terminators inside the file). Neither with the repository being inited on Linux, nor on Wine. Not even if I modified the code to use `"wb"`, i.e. output LF.

**Comment by h-vetinari** [comment]
Fixed by #2192
